controllers/topology/internal/extensions/patches/patch_test.go

```diff
@@ -0,0 +1,369 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package patches
+
+import (
+	"testing"
+
+	. "github.com/onsi/gomega"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"sigs.k8s.io/cluster-api/controllers/topology/internal/contract"
+)
+
+func TestCopySpec(t *testing.T) {
```

Comment:
Similar to engine.go, I've only added one test for copySpec. If preferred I can duplicate the test cases for the 3 patch funcs. 

I thought testing copySpec covers the most cases and the 3 other funcs are transitively tested by the tests in engine_test.go